File: solutions_1673486_1/java/Kraken/R1A.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;

public class R1A {
public static void main(String[] args) {
		String prblm="A"; boolean fl=true;
		String filein=prblm+"-"+((fl)?"large":"small")+".in.txt";
		String fileout=prblm+"-"+((fl)?"large":"small")+".out.txt";
		try {
			BufferedReader fr= new BufferedReader(new FileReader(filein));
			BufferedWriter fw= new BufferedWriter(new FileWriter(fileout));
			String s=fr.readLine();
			int T=Integer.parseInt(s);
			for (int i = 1; i <= T; i++) {
				s=fr.readLine();
				String[] tok=s.split(" ");
				int A=Integer.parseInt(tok[0]);
				int B=Integer.parseInt(tok[1]);
				s=fr.readLine();
				tok=s.split(" ");
				double emin=2+B;
				double prc=1;
				for (int j = 0; j < A; j++) {
					double pr=Double.parseDouble(tok[j]);
					prc*=pr;
					double ev=(A-j-1)+(B-j-1)+(B+1)*(1-prc)+1;
					emin=Math.min(emin, ev);
				}
				System.out.println(emin);
				fw.write("Case #"+i+": "+ emin  +"\n");
			}
			fr.close();
			fw.close();
		} catch (Exception e) {			
			e.printStackTrace();
		}		
	}
}

File: solutions_1673486_0/Python/littleeyes/r1_a.py
def r1_a():
	r = open('r1_a.in', 'r')
	w = open('r1_a.out', 'w')
	T = eval(r.readline())
	for c in range(T):
		line = r.readline().split(' ')
		a = eval(line[0])
		b = eval(line[1])
		p = r.readline().split(' ')
		for x in range(a):
			p[x] = eval(p[x])
		val = 1e9*1.0
		# case 1.
		sp = 1.
		for i in range(a):
			sp *= p[i]
		nval = sp*(b-a+1) + (1-sp)*(b-a+1+b+1)
		val = min(val, nval)

		# case 2.
		for k in range(1, a):
			sp = 1.
			for i in range(a-k):
				sp *= p[i]
			nval = sp*(b-a+k+1) + (1-sp)*(b-a+k+1+b+1) + k
			val = min(val, nval)
		nval = a+b+1
		val = min(val, nval)
		
		# case 3.
		nval = 1+b+1
		val = min(val, nval)

		xline = "Case #%d: %f"%(c+1,val)
		print xline
		w.write(xline+'\n')

if __name__ == '__main__':
	r1_a()
